Assets/Scripts/SS3D/Systems/Inventory/Items/ItemActor.cs

```diff
@@ -303,16 +310,107 @@ public void GenerateNewIcon()
             {
                 Texture2D texture = RuntimePreviewGenerator.GenerateModelPreviewWithShader(this.transform,
             Shader.Find("Legacy Shaders/Diffuse"), null, 128, 128, true, true);
-                _sprite = Sprite.Create(texture, new Rect(0, 0, texture.width, texture.height), new Vector2(0.5f, 0.5f), 100);
-                _sprite.name = transform.name;
+                item.Sprite = Sprite.Create(texture, new Rect(0, 0, texture.width, texture.height), new Vector2(0.5f, 0.5f), 100);
+                item.Sprite.name = transform.name;
             }
             catch (NullReferenceException)
             {
                 Debug.LogError("Null reference exception, reverting to default sprite for item " + name + ".");
             }
 
         }
+        #endregion
+
+        #region Item
+        public class Item
+        {
+            private ItemId _itemId;
+
+            /// <summary>
+            /// The item's name in the UI
+            /// </summary>
+            private string _name;
+
+            /// <summary>
+            /// The item's relative weight in kilograms.
+            /// </summary>
+            private float _weight;
+
+            /// <summary>
+            /// The amount of slots the item will take in a container
+            /// </summary>
+            private Vector2Int _size;
+
+            /// <summary>
+            /// The sprite that is shown in the container slot
+            /// </summary>
+            private Sprite _sprite;
+
+            /// <summary>
+            /// The list of characteristics this Item has
+            /// </summary>
+            private List<Trait> _traits;
+
+            /// <summary>
+            /// The container the item is currently stored on
+            /// </summary>
+            private Container _container;
+
+            public ItemId ItemId
+            {
+                get => _itemId;
+                set => _itemId = value;
+            }
+
+            public string Name
+            {
+                get => _name;
+                set => _name = value;
+            }
+
+            public float Weight
+            {
+                get => _weight;
+                set => _weight = value;
+            }
+
+            public Vector2Int Size
+            {
+                get => _size;
+                set => _size = value;
+            }
+
+            public Sprite Sprite
+            {
+                get => _sprite;
+                set => _sprite = value;
+            }
+
+            public List<Trait> Traits
+            {
+                get => _traits;
+                set => _traits = value;
+            }
+
+            public Container Container
```

Comment:
this one we need to be careful. it wasn't synced properly before because it updated using an RPC, so late joinin client would have the wrong container value when joining. We may need to make Container a syncvar or something (not this one but the one in itemActor)